internal/cache/controller_obj_cache.go

```diff
@@ -2763,14 +2802,99 @@ func checkAndSetCloudType(client *clients.AviClient) bool {
+	// Donot check for labels in usable networks if a vipNetwork is provided by the user.
+	// In this case, the vipNetwork provided by the user will be used.
+	// 1. Prioritize user input vipetworkList, skip marker based selection if provided.
+	// 2. If not provided, check for markers in ipam's usable networks.
+	// 3. If marker based usable network is not available, keep vipNetworkList empty.
+	// 4. vipNetworkList can be empty only in WCP usecases, for all others, mark invalid configuration.
+
+	// 1. User input
+	if vipList, _ := lib.GetVipNetworkListEnv(); len(vipList) > 0 {
+		lib.SetVipNetworkList(vipList)
+		return true
+	}
+
+	// 2. Marker based (only advancedL4)
+	markerNetworkFound := ""
+	if lib.GetAdvancedL4() && ipamRefUri != nil {
+		// Using clusterID for advl4.
+		clusterName := lib.GetClusterID()
+		ipam := models.IPAMDNSProviderProfile{}
+		ipamRef := strings.SplitAfter(*ipamRefUri, "/api/")
+		if err := lib.AviGet(client, "/api/"+ipamRef[1], &ipam); err != nil {
+			utils.AviLog.Errorf("Get uri %v returned err %v", ipamRef, err)
+			return false
+		}
+
+		usableNetworkNames := []string{}
+		for _, usableNetwork := range ipam.InternalProfile.UsableNetworks {
+			networkRefName := strings.Split(*usableNetwork.NwRef, "#")
+			usableNetworkNames = append(usableNetworkNames, networkRefName[1])
+		}
+
+		if len(usableNetworkNames) == 0 {
+			utils.AviLog.Errorf("No usable network configured in configured cloud's ipam profile.")
+			return false
+		}
+
+		// TODO: pagination
+		uri := "/api/network/?include_name&page_size=100&name.in=" + strings.Join(usableNetworkNames, ",")
+		result, err := lib.AviGetCollectionRaw(client, uri)
+		if err != nil {
+			utils.AviLog.Errorf("Get uri %v returned err %v", uri, err)
+			return false
+		}
+
+		elems := make([]json.RawMessage, result.Count)
+		err = json.Unmarshal(result.Results, &elems)
+		if err != nil {
+			utils.AviLog.Errorf("Failed to unmarshal data, err: %v", err)
+			return false
+		}
+
+		for _, elem := range elems {
+			network := models.Network{}
+			err = json.Unmarshal(elem, &network)
+			if err != nil {
+				utils.AviLog.Errorf("Get uri %v returned err %v", uri, err)
+				return false
+			}
+
+			if len(network.Markers) == 1 &&
+				*network.Markers[0].Key == lib.ClusterNameLabelKey &&
+				len(network.Markers[0].Values) == 1 &&
+				network.Markers[0].Values[0] == clusterName {
+				markerNetworkFound = *network.Name
+				utils.AviLog.Infof("Marker configuration found in usable network. Using %s as vipNetworkList.", markerNetworkFound)
+				break
+			}
+		}
+
+		if markerNetworkFound != "" {
+			lib.SetVipNetworkList([]akov1alpha1.AviInfraSettingVipNetwork{{
+				NetworkName: markerNetworkFound,
+			}})
+			return true
+		}
+
+	}
+
+	// 3. Empty VipNetworkList (only valid for WCP)
+	if markerNetworkFound == "" {
+		lib.SetVipNetworkList([]akov1alpha1.AviInfraSettingVipNetwork{})
+		return true
+	}
+
+	return false
```

Comment:
Would we ever reach here ? 
if markerNetworkFound is not empty then it would be handled in the if block 
if lib.GetAdvancedL4() && ipamRefUri != nil {
otherwise it would be handled in 
if markerNetworkFound == "" {